File: LeetCode/leetcode 7.py
class Solution:
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        s = str(x)
        n = ""
        if s[0] == "-":
            n = "-"
            s = s[1:]
        ans = int(n + s[::-1])
        if ans < -2**31 or ans > 2**31 - 1:
            return 0
        return ans

File: LeetCode/leetcode 7.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
   public:
	int reverse(int x) {
		long long x2 = x;
		string n = "";
		if (x2 < 0) {
			n = "-";
			x2 = -x2;
		}
		string s = to_string(x2);
		std::reverse(s.begin(), s.end());
		long long ans = stoll(n + s);
		if (ans < -2147483648LL || ans > 2147483647LL) {
			return 0;
		}
		return ans;
	}
};
